### PR TITLE
[v1.2][ISSUE-402] publish deltatuner to pypi

### PR DESCRIPTION
**Pull request by zhouyu5**
### What changes were proposed in this pull request?

Update the package and publish to pypi workflow

### Why are the changes needed?

For integration with third party internally or externally.

### How was this patch tested?

It is tested by package and uploading, see [here ](https://pypi.org/manage/project/deltatuner/releases/)for final package
The actions run details can be found [here](https://github.com/zhouyu5/e2eAIOK/actions/runs/6611652126/job/17956275025).

### CODE REVIEW

File: .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml [hunk 1]
```diff
@@ -2,41 +2,38 @@ name: Publish E2EAIOK Deltatuner nightly to PyPI
...
         nightly_version=${release_version}b${nightly_build_date}
-        echo $nightly_version > version
+        echo $nightly_version > deltatuner/version
         python3 setup.py sdist
 
     - name: Upload
-      uses: pypa/gh-action-pypi-publish@master
+      uses: pypa/gh-action-pypi-publish@release/v1
       with:
-        password: ${{ secrets.PYPI_API_TOKEN_PYRECDP }}
```
Comment (xuechendi) [inline .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml:41]: is it publishable without token?

**Review by xuechendi — APPROVED** [review]
Codes LGTM, please check the token question

File: .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml (hunk 1 above)
Comment (zhouyu5) [reply .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml:41]: @xuechendi Yes, I use the guide here: [Publishing to PyPI with a Trusted Publisher](https://docs.pypi.org/trusted-publishers/). It is a recommend way provided by PyPI and GitHub actions. I paste the introduction here:

> "Trusted publishing" is our term for using the [OpenID Connect (OIDC)](https://openid.net/connect/) standard to exchange short-lived identity tokens between a trusted third-party service and PyPI. This method can be used in automated environments and eliminates the need to use username/password combinations or manually generated API tokens to authenticate with PyPI when publishing.

**Comment by zhouyu5** [comment]
@tianyil1 Please have a review and feel free to leave your comments.

File: e2eAIOK/deltatuner/deltatuner/deltatuner.py [hunk 2]
```diff
@@ -12,9 +12,11 @@
 
 SUPPORTED_ALGO = ["auto", "lora", "ssf"]
 
-def optimize(model, tokenizer, algo: str="auto", adapter_name: str="default", deltatuning_args: DeltaTunerArguments=None) -> DeltaTunerModel:
+def optimize(model, tokenizer, adapter_name: str="default", deltatuning_args: DeltaTunerArguments=None, **kwargs) -> DeltaTunerModel:
     if deltatuning_args is None:
-        deltatuning_args = DeltaTunerArguments()
+        deltatuning_args = DeltaTunerArguments(**kwargs)
+
+    algo = deltatuning_args.algo
```
Comment (tianyil1) [inline e2eAIOK/deltatuner/deltatuner/deltatuner.py:19]: I suggested using the kwargs.pop("key_name", None) method if the "key_name" did not appear in the kwargs

**Review by tianyil1 — COMMENTED** [review]
I have left some comments. Please have a check.

File: .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml [hunk 3]
```diff
@@ -2,41 +2,38 @@ name: Publish E2EAIOK Deltatuner nightly to PyPI
...
 
-permissions:
-  contents: read
-  packages: write
 
 jobs:
-  e2eaiok-recdp-nightly-python-pypi:
-    runs-on: self-hosted
+  e2eaiok-dtuner-nightly-python-pypi:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (zhouyu5) [reply .github/workflows/e2eaiok_deltatuner_nightly_pypi.yml:14]: @tianyil1  It's just for package, not the integration test. Integration test is still on "self-hosted".

File: e2eAIOK/deltatuner/deltatuner/deltatuner.py (hunk 2 above)
Comment (zhouyu5) [reply e2eAIOK/deltatuner/deltatuner/deltatuner.py:19]: kwargs is used to initilize deltatuning_args, suppose will not conflict with other args.

**Pull request merged**